spec-tests/src/test/java/org/candlepin/spec/content/ConditionalContentSpecTest.java

```diff
@@ -0,0 +1,759 @@
+            .isEqualTo(conditionalContent1.getType());
+        assertThat(getContentRepoType(cert, conditionalContent2.getId()))
+            .isEqualTo(conditionalContent2.getType());
+        assertThat(getContentRepoType(cert, conditionalContent3.getId())).isNull();
+    }
+
+    @Test
+    public void shouldNotIncludeConditionalContentWithoutTheRequiredProducts() throws Exception {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.random());
+        String ownerKey = owner.getKey();
+        ProductDTO reqProd1 = adminClient.ownerProducts()
+            .createProductByOwner(ownerKey, Products.randomEng());
+        ProductDTO reqProd2 = adminClient.ownerProducts()
+            .createProductByOwner(ownerKey, Products.randomEng());
+        ProductDTO reqProd3 = adminClient.ownerProducts()
+            .createProductByOwner(ownerKey, Products.randomEng());
+
+        // Create our dependent provided product, which carries content sets -- each of which of which
+        // requires one of the provided products above
+        ProductDTO dependentProvProd = adminClient.ownerProducts()
+            .createProductByOwner(ownerKey, Products.random());
+        ContentDTO conditionalContent1 = adminClient.ownerContent().createContent(ownerKey, Content.random()
+            .modifiedProductIds(Set.of(reqProd1.getId())));
+        ContentDTO conditionalContent2 = adminClient.ownerContent().createContent(ownerKey, Content.random()
+            .modifiedProductIds(Set.of(reqProd2.getId())));
+        ContentDTO conditionalContent3 = adminClient.ownerContent().createContent(ownerKey, Content.random()
+            .modifiedProductIds(Set.of(reqProd3.getId())));
+        adminClient.ownerProducts()
+            .addContent(ownerKey, dependentProvProd.getId(), conditionalContent1.getId(), true);
+        adminClient.ownerProducts()
+            .addContent(ownerKey, dependentProvProd.getId(), conditionalContent2.getId(), true);
+        adminClient.ownerProducts()
+            .addContent(ownerKey, dependentProvProd.getId(), conditionalContent3.getId(), true);
+
+        ProductDTO dependentProd = adminClient.ownerProducts()
+            .createProductByOwner(ownerKey, Products.random()
+            .providedProducts(Set.of(dependentProvProd)));
+        adminClient.owners().createPool(ownerKey, Pools.random(dependentProd));
+
+        // Bind to the dependent subscription without being entitled to any of the required products
+        ConsumerDTO consumer = adminClient.consumers().createConsumer(Consumers.random(owner)
+            .facts(Map.of("system.certificate_version", "1.0")));
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+        JsonNode ents = consumerClient.consumers().bindProduct(consumer.getUuid(), dependentProd.getId());
+        String cert = getCertFromEnt(ents.get(0));
+        assertThat(getContentRepoType(cert, conditionalContent1.getId())).isNull();
```

Comment:
For one-liners like this, it might make sense to just use the junit base assertions for equality/null checks.

Reply:
Yeah, that makes sense. Updated throughout